Framework/Geometry/inc/MantidGeometry/Rendering/GeometryHandler.h

```diff
@@ -13,103 +16,88 @@ namespace Geometry {
 class IObjComponent;
 class ObjComponent;
 class CSGObject;
+namespace detail {
+class Renderer;
+class GeometryTriangulator;
+}
 
 /**
-   \class GeometryHandler
-   \brief Place holder for geometry triangulation and rendering.
-   \author Srikanth Nagella
-   \date July 2008
-   \version 1.0
+\class GeometryHandler
+\brief Handles rendering of all object Geometry.
+\author Lamar Moore
+\date December 2017
 
-   This is an abstract class for handling geometry primitives.
+Handles the rendering of all geometry types in Mantid.
 
-   Copyright &copy; 2008 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
-   National Laboratory & European Spallation Source
+Copyright &copy; 2008 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+National Laboratory & European Spallation Source
 
-   This file is part of Mantid.
+This file is part of Mantid.
 
-   Mantid is free software; you can redistribute it and/or modify
-   it under the terms of the GNU General Public License as published by
-   the Free Software Foundation; either version 3 of the License, or
-   (at your option) any later version.
+Mantid is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3 of the License, or
+(at your option) any later version.
 
-   Mantid is distributed in the hope that it will be useful,
-   but WITHOUT ANY WARRANTY; without even the implied warranty of
-   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-   GNU General Public License for more details.
+Mantid is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
 
-   You should have received a copy of the GNU General Public License
-   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-   File change history is stored at: <https://github.com/mantidproject/mantid>
+File change history is stored at: <https://github.com/mantidproject/mantid>
 */
 class MANTID_GEOMETRY_DLL GeometryHandler {
 private:
   static Kernel::Logger &PLog; ///< The official logger
 
 protected:
-  IObjComponent *ObjComp; ///< ObjComponent that uses this geometry handler
-  CSGObject *Obj;         ///< Object that uses this geometry handler
-  bool boolTriangulated;  ///< state of the geometry triangulation
-  bool
-      boolIsInitialized; ///< state of the geometry initialization for rendering
+  std::unique_ptr<detail::Renderer> m_renderer;
+  std::shared_ptr<detail::ShapeInfo> m_shapeInfo;
+  std::unique_ptr<detail::GeometryTriangulator> m_triangulator;
```

Comment:
Review use of smart pointers. If possible move to stack-allocated objects. You mentioned that these could be null, in which case leave as they are.